### PR TITLE
fix: datarace in datarace test

### PR DESCRIPTION
**Pull request by shaneutt**
This fixes a really ironic bug found in our unit tests.

### CODE REVIEW

File: pkg/util/debug_logging_test.go [hunk 1]
```diff
@@ -132,10 +132,18 @@ func TestDebugLoggerThreadSafety(t *testing.T) {
 	log := MakeDebugLoggerWithReducedRedudancy(buf, &logrus.JSONFormatter{}, 0, time.Minute*30)
 
 	// spam the logger concurrently across several goroutines to ensure no dataraces
+	wg := &sync.WaitGroup{}
 	for i := 0; i < 100; i++ {
-		go func() { log.Debug("unique") }()
+		wg.Add(1)
```
Comment (mflendrich) [inline pkg/util/debug_logging_test.go:137]: nit: just `Add` once and put `100` in a constant

File: pkg/util/debug_logging_test.go [hunk 2]
```diff
@@ -132,10 +132,18 @@ func TestDebugLoggerThreadSafety(t *testing.T) {
...
+		go func() {
+			defer wg.Done()
+			log.Debug("unique")
+		}()
 	}
+	wg.Wait()
 	assert.Contains(t, buf.String(), "unique")
+	lines := strings.Split(buf.String(), "\n")
+	assert.True(t, len(lines) < 100)
```
Comment (mflendrich) [inline pkg/util/debug_logging_test.go:146]:
```suggestion
	// Ensure that _some_ lines have been stifled. The actual number is not deterministic.
	assert.True(t, len(lines) < 100)
```

**Review by mflendrich — APPROVED** [review]
conditional lgtm if all suggestions get applied

File: pkg/util/debug_logging_test.go (hunk 1 above)
Comment (shaneutt) [reply pkg/util/debug_logging_test.go:137]: changed: 73ad9ebd8bd66adb1e9266510e8f97a549f5e159

**Pull request merged**